### ISSUE TITLE
Disable select only for certain rows (EDIT: and columns)

### ISSUE THREAD

**Issue by tomasreigl** [rstudio/DT#201]
Hello everybody,
I would like to ask, if it is possible to disable select only for some rows. I "do" it with a little hack, but maybe there is some smarter way to do it.

server:

``` r
library(shiny)
library(DT)
shinyServer(function(input, output, session) {
    output$foo = DT::renderDataTable( iris )
    proxy = dataTableProxy('foo')
    # observe diabling desired rows
    myClear <- observe({
        # vector of "disabled" rows
        deadList <- c(2,3)
        i_deadList <- input$foo_rows_selected %in% deadList
        if (any(i_deadList)) {
            selectRows(proxy, input$foo_rows_selected[!i_deadList])
        }
    })
})
```

ui:

``` r
library(shiny)
fluidPage(
    DT::dataTableOutput('foo')
)
```

Thank you very much.

EDIT:
I also wonder if there is any way to disable selection functionality for certain column, so if user clicks on a cell in this column, the row is not selected.
Thank you again :)

**Comment by yihui** [comment]
Sounds like a nice feature to have. I'll consider it in a future version of DT. Thanks!

**Comment by jonthegeek** [comment]
I'd like to add my vote for this. It would definitely be useful!

**Comment by incivile** [comment]
I'm using a bit smarter code right now to do this, so maybe it will help somebody. And you can disable selecting even for columns with this example:

```
library(shiny)
library(DT)

shinyApp(
    ui = fluidPage(
        DT::dataTableOutput('foo'),
        verbatimTextOutput('selected_rows')
    ),
    
    server = function(server, input, output) {
        
        output$foo <- DT::renderDataTable({
            
            my_data <- iris
            ### disable row selecting for some rows ###
            disable <- rep(c(T, F), length = nrow(my_data)) # T - not able to select, F - selecting disabled
            regexp_disable <- rep(c("OK", "OK", "notOK"), length = nrow(my_data))
            my_data <- cbind(regexp_disable, disable, my_data)
            # in the end you can use any column in the table to do this or you can hide these helping ones
            
            ### disable row selecting for some columns ###
            my_options <- list()
            my_cols <- c(1, 3) # disable selecting for 2nd and 4th column
            my_options$columnDefs[[length(my_options$columnDefs) + 1]] <- list(className = 'no_select', targets = my_cols)
            # 'length(my_options$columnDefs) + 1' is used, because there could be some columnDefs definitions before
            
            datatable(my_data,
                      rownames = F,
                      options = my_options,
                      callback = JS(paste0("
                                    /* disable row selecting for some rows */
                                     table.on('click', 'td', function() {
                                        var td = $(this), row = table.row(td.closest('tr'));
                                        if (RegExp('not').test(table.cell(row, 0).data()) ||
                                            table.cell(row, 1).data()) {
                                            $(td.node()).removeClass('selected');
                                        }
                                     });
                                    /* disable row selecting for some columns */
                                     table.on('click', 'td.no_select', function() {
                                        var td = $(this), row = table.row(td.closest('tr'));
                                        if ($(td.node()).hasClass('selected')) {
                                            $(td.node()).removeClass('selected');
                                        }
                                     });"))
            )
                      
        })
        
        output$selected_rows <- renderText({
            input$foo_rows_selected
        })
    }
)
```

**Comment by lisovyk** [comment]
Is it implemented already, maybe?

**Comment by yihui** [comment]
@lisovyk Not yet (by me). Sorry.

**Comment by stanstrup** [comment]
This looks like what I need but doesn't work anymore it seems.

**Comment by dwhdai** [comment]
This seems to be supported by [datatables](https://editor.datatables.net/examples/inline-editing/columns.html). Any chance this will be implemented into DT?

**Comment by stla** [comment]
This is implemented now.

**Comment by Merchant93** [comment]
@stla does the feature have a vignette yet? I can't seem to find it anywhere.

**Comment by kikegoni** [comment]
Is it any update on that?

**Comment by stla** [comment]
See the doc of the `selection` argument in `?DT::datatable`.

**Comment by stanstrup** [comment]
I don't see anything in the documentaion about this. It is only explained how you do preselection.

**Comment by stla** [comment]
Sorry, I did a confusion with the disabling of editable rows.

**Comment by shrektan** [comment]
I have implemented a draft version for this feature in #793. Feedback is welcome.

(You can test it by `remotes::install_github('rstudio/DT#793')`)
